src/SwaggerProvider/SwaggerProvider.fs

```diff
@@ -19,17 +20,38 @@ type public SwaggerProvider(cfg : TypeProviderConfig) as this =
     let tempAsm = ProvidedAssembly tempAsmPath
 
     let t = ProvidedTypeDefinition(asm, ns, "SwaggerProvider", Some typeof<obj>, IsErased = false)
-    let parameters = [ProvidedStaticParameter("Schema", typeof<string>)]
+
+    let parameters = [ProvidedStaticParameter("Schema", typeof<string>)
+                      ProvidedStaticParameter("Headers", typeof<string>,"")]
 
     do
         t.DefineStaticParameters(
             parameters=parameters,
-            instantiationFunction=(fun typeName args ->
-                let schemaPath = args.[0] :?> string
+            instantiationFunction = (fun typeName args ->
+                let h = args.[1] :?> string
+                let headers = h.Split(';') |> Seq.filter (fun f -> f.Contains(",")) |> Seq.map (fun e -> 
+                    let pair = e.Split(',')
+                    (pair.[0],pair.[1])
+                    )                    
+
+                let schemaPathRaw = args.[0] :?> string
+
+                let schemaPath = 
+                    match schemaPathRaw.StartsWith("http", true, null) with
+                    | true -> 
+                        let root =  __SOURCE_DIRECTORY__
+                        let swaggerFilePath = root + @"/../../tests/SwaggerProvider.Tests/Schemas/swaggerSchema.json"
```

Comment:
Hi Sergey, I'm helping @skovlund  to get another improved pull request sent to you today, with fixes. Thanks for the feedback.